counterpartylib/lib/transaction.py

```diff
@@ -487,4 +491,199 @@ def construct (db, tx_info, encoding='auto',
 
     return unsigned_tx_hex
 
+
+"""
+Client side checking: never uses database or backend.
+"""
+
+def raw_unpack_data(data_hex):
+    from counterpartylib.lib.api import API_TRANSACTIONS
+
+    try:
+        data = binascii.unhexlify(data_hex)
+        message_type_id = struct.unpack(config.TXTYPE_FORMAT, data[:4])[0]
+        message = data[4:]
+
+        for message_type in API_TRANSACTIONS:
+            message_module = sys.modules['counterpartylib.lib.messages.{}'.format(message_type)]
+            if message_type_id == message_module.ID:
+                params = {
+                    'message': message_type
+                }
+                if hasattr(message_module, 'FORMAT'):
+                    format = message_module.FORMAT(message) if callable(message_module.FORMAT) else message_module.FORMAT
+                    unpacked = struct.unpack(format, message)
+                    for p in range(len(unpacked)):
+                        param_name = message_module.FORMAT_DESCRIPTION[p]
+                        params[param_name] = unpacked[p]
+                    return params
+    except struct.error as e:
+        raise exceptions.UnpackError(e)
+
+    raise exceptions.UnpackError("Invalid data")
```

Comment:
if there is no return before, that means if there is no corresponding message ID.